TCS34725.h

```diff
@@ -206,16 +206,20 @@ class TCS34725_
         x |= t;
         return x;
     }
+    
+    void readCRGB(RawData *crgb)
+    {
 
-private:
+        wire->beginTransmission(I2C_ADDR);
+        wire->write(COMMAND_BIT | (uint8_t) Reg::CDATAL);
+        wire->endTransmission();
 
-    void update()
-    {
-        raw_data.r = read16(Reg::RDATAL);
-        raw_data.g = read16(Reg::GDATAL);
-        raw_data.b = read16(Reg::BDATAL);
-        raw_data.c = read16(Reg::CDATAL);
-    }
+        wire->requestFrom(I2C_ADDR, sizeof(RawData));
+        for (uint8_t i = 0; i < sizeof(RawData); i++)
+          *((uint8_t *) crgb+i) = wire->read();
+    }    
+
+private:
```

Comment:
If you use `union` you can change here like:

```C++
    void update()
    {
        wire->beginTransmission(I2C_ADDR);
        wire->write(COMMAND_BIT | (uint8_t) Reg::CDATAL);
        wire->endTransmission();
        wire->requestFrom(I2C_ADDR, sizeof(RawData));
        for (uint8_t i = 0; i < sizeof(RawData); i++)
            raw_data.raw[i] = wire->read();

        if (raw_data.c == 0) clr.r = clr.g = clr.b = 0;
        else
        {
            clr.r = pow((float)raw_data.r / (float)raw_data.c, scaling) * 255.f;
            clr.g = pow((float)raw_data.g / (float)raw_data.c, scaling) * 255.f;
            clr.b = pow((float)raw_data.b / (float)raw_data.c, scaling) * 255.f;
            if (clr.r > 255.f) clr.r = 255.f;
            if (clr.g > 255.f) clr.g = 255.f;
            if (clr.b > 255.f) clr.b = 255.f;
        }
    }
```

Reply:
I agree for the readout, still I would propose to move the color calculation to the function `const Color& color()` itself. If you just want raw data, this calculation just uses unnecessary processor cycles without benefit and it is a transparent change without interface adaptation.